### PR TITLE
Fixed the error during dry run

### PR DESCRIPTION
**Pull request by xgreenx**
Because of race condition it was possible that dry run will fail. I moved the check to production, where we really need to require this behavior. But in the case of the dry run we only care about previous block existence.

## Checklist
- [x] New behavior is reflected in tests

### Before requesting review
- [x] I have reviewed the code myself

### CODE REVIEW

File: crates/services/consensus_module/poa/src/service.rs [hunk 1]
```diff
@@ -359,8 +359,6 @@ where
             consensus: seal,
         };
 
-        block.entity.header().time();
-
```
Comment (rymnc) [inline crates/services/consensus_module/poa/src/service.rs:363]: nice catch

**Review by AurelienFT — APPROVED** [review]

**Review by acerone85 — APPROVED** [review]
Did not go in depth with the review, but I can follow the main idea behind the fix.
LGTM

**Review by AurelienFT — DISMISSED** [review]

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

**Review by AurelienFT — APPROVED** [review]

File: crates/services/producer/src/block_producer/tests.rs [hunk 2]
```diff
@@ -649,27 +649,33 @@ mod dry_run {
     }
 
     #[tokio::test]
-    async fn dry_run__errors_early_if_height_is_lower_than_chain_tip() {
```
Comment (MitchTurner) [inline crates/services/producer/src/block_producer/tests.rs:652]: I did a quick scan but couldn't see a test for this on the produce or produce_predefined. Maybe I missed it. Do you think we need that coverage?

Comment (xgreenx) [reply crates/services/producer/src/block_producer/tests.rs:652]: We have test case for `BlockHeightShouldBeHigherThanPrevious` error. 

image

But after introduction of the state rewind feature, I don't think that `BlockHeightShouldBeHigherThanPrevious` error makes a lot of sense. Basically importer already is responsible and upgradable executor do this kind of check inside, some maybe we can just remove it from the block producer

**Pull request merged**